augmentation/augmentation_impl.hpp

```diff
@@ -0,0 +1,74 @@
+/**
+ * @file augmentation_impl.hpp
+ * @author Kartik Dutt
+ * 
+ * Implementation of Augmentation class for augmenting data.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+// Incase it has not been included already.
+#include "augmentation.hpp"
+
+#ifndef MODELS_AUGMENTATION_IMPL_HPP
+#define MODELS_AUGMENTATION_IMPL_HPP
+
+template<typename DatasetType>
+void Augmentation::Transform(DatasetType& dataset,
+                             const size_t datapointWidth,
+                             const size_t datapointHeight,
+                             const size_t datapointDepth)
+{
+  // Initialize the augmentation map.
+  std::unordered_map<std::string, void(*)(DatasetType&,
+       size_t, size_t, size_t, std::string&)> augmentationMap;
+
+  size_t i = 0;
+  if (this->HasResizeParam())
+  {
+    this->ResizeTransform(dataset, datapointWidth, datapointHeight,
+        datapointDepth, augmentations[0]);
+    i++;
+  }
```

Comment:
So I am guessing that there should only be 1 resize transform. Maybe we should throw an error if we find 2.

Reply:
I changed it to allow multliple resize transforms. I hope this change makes sense.